### PR TITLE
Add DasCustodySync unit test

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

Add `DasCustodySync` test which is 'slightly integrational' cause the testing is integrated with `DataColumnCustodyImpl`

Tests check if the sync is complete, and the number of read requests to 
- DataColumnDb
- Block db

Later we will need to check the speed of sync as well. Having 0 latency of the `SidecarRetriever` the sync should normally be completed immediately. 

Currently we are supposing max db reads to be `30` per slot on average. We need to reduce this number in future

The test `nonFinalizationShouldNotPreventSyncingAndOverloadDB` is disabled now as it fails with both assersions: db reads (which grow every slot) and being able to sync

The #117 refactor should help to resolve the above issues

### CODE REVIEW

File: ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        .orElse(Collections.emptyList());
+  }
+
+  public int getCustodyPeriodSlots() {
+    int custodyPeriodEpochs = config.getCustodyRequirement();
+    int epochSlots = spec.slotsPerEpoch(UInt64.ZERO);
+    return custodyPeriodEpochs * epochSlots;
+  }
+
+  public UInt64 getMinCustodySlot() {
```
Comment (zilm13) [inline ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java:123]: it should be whole epochs as per spec

File: ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+import tech.pegasys.teku.spec.datastructures.blobs.versions.eip7594.DataColumnSidecar;
+import tech.pegasys.teku.spec.datastructures.blocks.BeaconBlock;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.BeaconBlockBody;
+import tech.pegasys.teku.spec.datastructures.state.Checkpoint;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.storage.api.FinalizedCheckpointChannel;
+
+@SuppressWarnings("unused")
+public class DasCustodyStand {
```
Comment (zilm13) [inline ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java:40]: Maybe call it `DasCustodyStub` in analogy with others?

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodySyncTest.java [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+    // on start we have 1000 uncustodied slots
+    custodyStand.setCurrentSlot(1000);
+    custodyStand.subscribeToSlotEvents(dasCustodySync);
+    dasCustodySync.start();
+
+    printAndResetStats();
+
+    for (int slot = startSlot + 1; slot <= startSlot + 1000; slot++) {
+      System.out.println("Processing slot #" + slot);
```
Comment (zilm13) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodySyncTest.java]: Looks like too much noise here, maybe move it per epoch below?

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodySyncTest.java [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+
+    printAndResetStats();
+
+    List<ColumnSlotAndIdentifier> missingColumns =
+        custodyStand.custody.retrieveMissingColumns().join();
+    assertThat(missingColumns).isEmpty();
+    assertAllCustodyColumnsPresent();
+  }
+
+  @Disabled
```
Comment (zilm13) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodySyncTest.java]: Could you please add comment why it's disabled

File: ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java (hunk 1 above)
Comment (Nashatyrev) [reply ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java:123]: I want slots :))

File: ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java (hunk 2 above)
Comment (Nashatyrev) [reply ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java:40]: It's not actually a 'stub'. It's the production `DasCustody` implementation + supporting classes and stubs packed in a single class. So it is kind of 'experimental stand' (стенд) or whatever else. Would be cool if you have any other appropriate term in mind

**Review by zilm13 — APPROVED** [review]
LGTM

File: ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java (hunk 1 above)
Comment (Nashatyrev) [reply ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java:123]: Yep, you are right, we need to round to the epoch boundary

**Pull request merged**